src/chainstate/coordinator/mod.rs

```diff
@@ -683,6 +1008,105 @@ impl<
         )
     }
 
+    fn process_exit_reward_cycle(
+        &mut self,
+        block_receipt: &StacksEpochReceipt,
+        block_hash: &BlockHeaderHash,
+        canonical_sortition_tip: &SortitionId,
+        exit_contract_id: &QualifiedContractIdentifier,
+    ) -> Result<(), Error> {
+        let current_block_height = block_receipt.header.burn_header_height as u64;
+        let current_reward_cycle = self
+            .burnchain
+            .block_height_to_reward_cycle(current_block_height)
+            .ok_or_else(|| DBError::NotFoundError)?;
+        let mut current_exit_at_rc = None;
+        let mut current_proposal = None;
+        let mut invalid_reward_cycles = vec![];
+        let stacks_block_id = StacksBlockId::new(&block_receipt.header.consensus_hash, &block_hash);
+        let parent_block_id = self
+            .chain_state_db
+            .get_parent(&stacks_block_id)
+            .expect("BUG: failed to get parent for processed block");
+
+        // look up parent in exit_at_reward_cycle_info table
+        let exit_info_opt =
+            SortitionDB::get_exit_at_reward_cycle_info(self.sortition_db.conn(), &parent_block_id)?;
+
+        if let Some(parent_exit_info) = exit_info_opt {
+            info!("RCPR: cc: parent block info is non-None");
+            // copy info from parent block
+            current_exit_at_rc = parent_exit_info.curr_exit_at_reward_cycle;
+            if parent_exit_info.block_reward_cycle == current_reward_cycle {
+                current_proposal = parent_exit_info.curr_exit_proposal;
+            }
+            invalid_reward_cycles = parent_exit_info
+                .invalid_reward_cycles
+                .iter()
+                .filter(|rc| **rc > current_reward_cycle)
+                .map(|rc| *rc)
+                .collect();
+
+            // We process the previous reward cycle as soon as we transition to the next one.
+            if parent_exit_info.block_reward_cycle < current_reward_cycle {
+                info!("RCPR: cc: ancestor - rc is greater than parent rc");
+                // Check if there is some proposal. If so, need to check for a veto.
+                if let Some(curr_exit_proposal) = parent_exit_info.curr_exit_proposal {
+                    let veto_passed = self.read_veto_state(
```

Comment:
To be clear, we only check the veto status for a reward cycle _once_, right?  i.e. only once a block gets mined whose parent is in a subsequent reward cycle?  And, we're checking to see if the parent's proposed exit reward cycle got vetoed?